plugins/parsers/avro/schema_registry.go

```diff
@@ -0,0 +1,61 @@
+package avro
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/linkedin/goavro/v2"
+	"net/http"
+)
+
+type SchemaAndCodec struct {
+	Schema string
+	Codec  *goavro.Codec
+}
+
+type SchemaRegistry struct {
+	url   string
+	cache map[int]*SchemaAndCodec
+}
+
+const (
+	schemaByID = "%s/schemas/ids/%d"
+)
+
+func NewSchemaRegistry(url string) *SchemaRegistry {
+	return &SchemaRegistry{url: url, cache: make(map[int]*SchemaAndCodec)}
+}
+
+func (sr *SchemaRegistry) getSchemaAndCodec(id int) (*SchemaAndCodec, error) {
+	if v, ok := sr.cache[id]; ok {
+		return v, nil
+	}
+	resp, err := http.Get(fmt.Sprintf(schemaByID, sr.url, id))
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var jsonResponse map[string]interface{}
+
+	err = json.NewDecoder(resp.Body).Decode(&jsonResponse)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This will make use of a limited-scope `err` variable
```suggestion
	if err := json.NewDecoder(resp.Body).Decode(&jsonResponse); err != nil {
		return nil, err
	}
```